provider/bidengine/pricing_test.go

```diff
@@ -545,6 +548,40 @@ func Test_ScriptPricingWritesJsonToStdin(t *testing.T) {
 	}
 }
 
+func Test_ScriptPricingFromScript(t *testing.T) {
+	const (
+		mockAPIResponse = `{"akash-network":{"usd":3.57}}`
+		addr            = "localhost:8080"
+		expectedPrice   = 67843138
+	)
+
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		_, err := fmt.Fprint(w, mockAPIResponse)
+		require.NoError(t, err)
+	})
+
+	go func() {
+		t.Error(http.ListenAndServe(addr, nil))
```

Comment:
It is best to use port `0` and get an ephemeral port in tests.

We should also shut down the server before returning.

Reply:
Done. Added `httptest` server which starts with port `0`